llvm/include/llvm/Support/PropertySetIO.h

```diff
@@ -230,6 +230,17 @@ class PropertySetRegistry {
     PropSet.insert({PropName, PropertyValue(PropVal)});
   }
 
+  void remove(StringRef Category, StringRef PropName) {
+    auto PropertySetIt = PropSetMap.find(Category);
+    if (PropertySetIt == PropSetMap.end())
+      return;
+    auto &PropertySet = PropertySetIt->second;
+    auto PropIt = PropertySet.find(PropName);
```

Comment:
Can we not call erase directly without checking if it exists first? It shouldn't make much of a difference in terms of performance, but that would save us a few lines of code.

Reply:
In testing I found that this function remove will sometimes be called in cases where `PropName (=="reqd_work_group_size_uint64_t")` is not in `PropertySet` and `erase` here does not work when the iterator is `end()`, so we can't remove the check.